[Devanshu Bisht] iP

devanshubisht opened the pull request:
# DukePro

> "Your mind is for having ideas, not holding them." – David Allen

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,

1. [Download it from here.](link)
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you 😉
   
And it is **FREE**!

### Features:
- [ ] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

devanshubisht committed:
finished level-2

devanshubisht committed:
finished level-3

devanshubisht committed:
finished level-5

devanshubisht committed:
finished level-6 and editted cases of delete/mark without any entry

devanshubisht committed:
editted errors with exceptions for empty description. added enum types. merged handle input with the previous cases of bye and list

devanshubisht committed:
Added Auto-Testing

devanshubisht committed:
Level-7 done

devanshubisht committed:
changes to git ignore

devanshubisht committed:
remove data

devanshubisht committed:
Finished Date-Time

devanshubisht committed:
editted format for exceptions

devanshubisht committed:
Added new classes

devanshubisht committed:
Better OOP

devanshubisht committed:
finishh

devanshubisht committed:
Add Graddle Support

devanshubisht committed:
Add JUnit testcase for DeadlineTaskTest & TaskListTest

devanshubisht committed:
Add JAR File

mingyuanc commented on src/main/java/devybot/DevyBot.java:
```diff
@@ -0,0 +1,42 @@
+
+import devybot.exceptions.DevyBotException;
+
+public class DevyBot {
```
Should there be javadocs written here?

mingyuanc commented on src/main/java/devybot/Parser.java:
```diff
@@ -0,0 +1,74 @@
+        }
+    }
+
+    public static void parse(String userInput, TaskList taskList) throws DevyBotException {
```
Should there be javadocs written here, as it is a rather important public method

mingyuanc commented on src/main/java/devybot/Parser.java:
```diff
@@ -0,0 +1,74 @@
+        CommandType commandType = getCommandType(wordsArray[0]);
+
+        switch (commandType) {
+            case TODO:
```
Should the case be at the same indentation level as the swtich?

mingyuanc commented on src/main/java/devybot/Parser.java:
```diff
@@ -0,0 +1,74 @@
+    }
+
+    public static void parse(String userInput, TaskList taskList) throws DevyBotException {
+        String[] wordsArray = userInput.split("\\s+");
```
good use of \\s+ to split the array with separator as space or multiple spaces

mingyuanc commented on src/main/java/devybot/tasks/Task.java:
```diff
@@ -0,0 +1,34 @@
+        this.isDone = false;
+    }
+
+    public boolean checkDone() {
```
Should this instead by isDone() or isMarked()? so that it sounds more like a boolean

mingyuanc commented on src/test/java/devybot/TaskListTest.java:
```diff
@@ -0,0 +1,50 @@
+    }
+
+    @Test
+    public void addTodoTask_validDescription_taskAdded() throws EmptyDescriptionException {
```
Good that all of your test cases method names follow the coding standard

mingyuanc reviewed (commented):
Overall pretty well done and clean code.
Could improve by adding javadocs on public method and separating logical units wihin a block with one blank line.

leawang0 commented on src/main/java/devybot/Parser.java:
```diff
@@ -0,0 +1,74 @@
+        }
+    }
+
+    public static boolean isRun() {
```
Perhaps a different function name would be better eg. `getRunStatus()`

leawang0 commented on src/main/java/devybot/Storage.java:
```diff
@@ -0,0 +1,108 @@
+import devybot.tasks.DeadlineTask;
+import devybot.tasks.EventTask;
+import devybot.tasks.Task;
+import devybot.tasks.TodoTask;
```
I like how you arranged your import statements nicely!

leawang0 commented on src/main/java/devybot/Storage.java:
```diff
@@ -0,0 +1,108 @@
+                Task loadedTask;
+
+                switch (taskType) {
+                    case "T":
```
identation level is not complying with our coding standard, perhaps need to fix

leawang0 commented on src/main/java/devybot/TaskList.java:
```diff
@@ -0,0 +1,164 @@
+        Ui.showMessage(outpString);
+    }
+
+    public void addDeadlineTask(String userInput) throws EmptyDescriptionException {
```
Perhaps should consider abstracting the logic for parsing and adding a task, right now you have duplicate code in both the `addDeadlineTask()` and `addEventTask()`, as well as in your `Storage` class

leawang0 commented on src/main/java/devybot/exceptions/DevyBotException.java:
```diff
@@ -0,0 +1,8 @@
+package devybot.exceptions;
+
+public class DevyBotException extends Exception {
+        public DevyBotException(String message) {
```
Identation level should be 1 tab apart, seems like there are 2 tabs

leawang0 commented on src/main/java/devybot/tasks/TodoTask.java:
```diff
@@ -0,0 +1,16 @@
+package devybot.tasks;
+public class TodoTask extends Task {
```
nit: add a newline between package declaration and class code would be nice

devanshubisht committed:
Add find function to find tasks with matching description

devanshubisht committed:
Include GUI capabilities

devanshubisht committed:
Remove unused packages

devanshubisht committed:
Add java assert in TaskList, Storage and DevyBot

TaskList, Storages and DevyBot have certain assumptions when adding new tasks,
saving files and returning the response.

Java assert helps to check for assumptions in the code.

I have identified areas to add assert such that the response to Main is never
empty. The new task created and being added to TaskList is never NULL. Storage
ensures that the directory and files are being created appropriately.

devanshubisht committed:
Refactor Code to CodingStandards

There are several classes with incorrect standards in the import section.
The switch statement violates the standard.

Such errors reflects poor code quality and difficulty to read.

Standardising the import section and refactoring the switch case clauses
to make the code aligned with the JavaCodingStandards.

devanshubisht committed:
Refactored Code to CodingStandards

There are several classes with incorrect standards in the import section.
The switch statement violates the standard.

Such errors reflects poor code quality and difficulty to read.

Standardising the import section and refactoring the switch case clauses
to make the code aligned with the JavaCodingStandards.

devanshubisht committed:
Merge branch 'branch-A-CodingStandard'

devanshubisht committed:
Refactor Code for better Code Quality

devanshubisht committed:
Merge pull request #1 from devanshubisht/branch-A-CodeQuality

Refactor Code for better Code Quality

devanshubisht committed:
Merge branch 'master' into branch-A-Assertions

devanshubisht committed:
Merge pull request #2 from devanshubisht/branch-A-Assertions

Add java assert in TaskList, Storage and DevyBot

devanshubisht committed:
Improve Code Quality in Storage

devanshubisht committed:
Add better search for partial description and dates

devanshubisht committed:
Edit DialogBox style to show all tasks

List command was getting truncated in the
DialogBox container.

This prevented users from viewing all tasks
in their list.

Enlarged the size of the DialogBox to allow users
to view all tasks.

devanshubisht committed:
Add UserGuide

devanshubisht committed:
Edit UserGuide

devanshubisht committed:
Add JavaDoc

devanshubisht committed:
Edit UserGuide

devanshubisht committed:
Increase width of UI

devanshubisht committed:
Add more JavaDocs

devanshubisht committed:
Edit JavaFX Version

devanshubisht committed:
Edit JavaFX Version
